### DISCUSSION TITLE
New daily (and yearly) plots

### DISCUSSION THREAD

**Discussion by jmtmp** [Nachtzuster/BirdNET-Pi#184] [Ideas]
I can share the code if anyone likes it.
I've been using it for a few months, currently I updated it according to the current daily_plot.py (daemon, colors, fonts)

![Combo light](https://github.com/user-attachments/assets/57aa2d8a-25cd-448f-a7f6-44e0b615977f)
![Combo-dark](https://github.com/user-attachments/assets/cc2dcc7e-ef94-4d20-a4ef-68b7c3f86c3c)

**Comment by lloydbayley** [comment]
That looks quite nice (and useful!). Since it takes up a little bit more space, how does it react wrapping/cut off-wise to extra-long bird names?

**Reply by jmtmp** [reply]
Text wrap is not implemented. For the czech names the table size is sufficient. 
Honestly, too many wrapped names were a one  impuls to made new charts.
If need be, there is option to implement it, or to try substitute long names by dots, or to adjust text size.

**Reply by lloydbayley** [reply]
Fair-point. The one that concerned me is our Sulphur-Crested Cockatoo. This name has the unfortunate habit of getting cut off just before the last four letters making it sound rude sometimes! :)

**Reply by alexbelgium** [reply]
Looks nice! Could you please share the code? Thanks !

**Reply by jmtmp** [reply]
Yes, here it is.
There are two plots - daily stat and yearly stat (in place of bottom observations plot)
[daily_plot.py.txt](https://github.com/user-attachments/files/16932459/daily_plot.py.txt)

**Reply by jmtmp** [reply]
By the way, when I look at the charts on the "DailyCharts" page on mobile, they surprisingly respond to gestures and are annoyingly redrawn

**Reply by alexbelgium** [reply]
They are redrawn at each detection which indeed leads to this behavior... Thanks for having shared I'll try them !

**Reply by jmtmp** [reply]
No, any left/right swipe changes the date

**Reply by alexbelgium** [reply]
Looks super nice :-) Encountered just minor clipping. Only remaining thing would be to align color codes with dark mode, I'll have a look

I've implemented it in the Birdnet-Pi homeassistant addon (with adding your name to the script) : https://github.com/alexbelgium/BirdNET-Pi/blob/new_daily_graph/scripts/daily_plot.py

![IMG_5291](https://github.com/user-attachments/assets/a1653176-60d1-48d4-8955-a42867a9839e)

**Comment by alexbelgium** [comment]
Proposal : wouldn't it be cleaner if the daily chart had no background for the species ?
However, to keep the background for the yearly chart or it could become confusing

![image](https://github.com/user-attachments/assets/91a59197-3ce8-44d0-b92a-b4d158ab9741)

**Comment by jmtmp** [comment]
Another, improved version of charts with
total detection count, to indicate uncommon or rare birds
<img width="1000" height="1064" alt="Combo-2025-09-18" src="https://github.com/user-attachments/assets/e74a444f-af2e-4429-8fe2-9699b9b793f0" />
